### PR TITLE
Add positional audio parameters

### PR DESCRIPTION
**Pull request by alexdevteam**
- Adds position and velocity to audio sources
- Showcases it in example

### CODE REVIEW

File: example/example.cpp [hunk 1]
```diff
@@ -1,33 +1,75 @@
...
 #include <chrono>
+#include <cmath>
 #include <fstream>
 #include <iostream>
 #include <thread>
 
 namespace impl {
 static constexpr int fail_code = 2;
+// TODO: Add parameters for each one of these constants [Use clap?]
+static constexpr float travel_circurference_radius = 2;
```
Comment (karnkaul) [inline example/example.cpp:12]: `2.0f` (literal should be float)

File: example/example.cpp [hunk 2]
```diff
@@ -1,33 +1,75 @@
...
+static constexpr float travel_angular_speed = 2;
+static constexpr bool loop_audio = true;
 
-int openAlTest(std::string const& wavPath, float gain) {
-	auto pcm = capo::PCM::fromFile(wavPath);
-	if (!pcm) { return impl::fail_code; }
+// Uniform circular motion formulas
+capo::Vec3 UCMPosition(float angularSpeed, float time, float radius) {
```
Comment (karnkaul) [inline example/example.cpp:17]: Is this going to be useful in a game engine?

File: example/example.cpp [hunk 3]
```diff
@@ -1,33 +1,75 @@
...
+
+		source.position(position);
+		source.velocity(velocity);
+
 		int const progress = static_cast<int>(20 * source.played() / sound.length());
-		while (progress > done) {
-			std::cout << '=' << std::flush;
-			++done;
+		if (progress > done) {
+			std::cout << "\r  ";
```
Comment (karnkaul) [inline example/example.cpp:69]: `\r`?

File: src/impl_al.hpp [hunk 4]
```diff
@@ -46,6 +47,7 @@ inline void onError(Error error) {
 inline bool alCheck() {
 	if (auto err = alGetError(); err != AL_NO_ERROR) {
 		onError(Error::eOpenALError);
+		std::cerr << err << std::endl;
```
Comment (karnkaul) [inline src/impl_al.hpp:50]: Why another `cerr`? `onError` is the customization point that `cerr`s by default; this way users have no way to suppress library output, and it occurs twice by default.

**Review by karnkaul — CHANGES_REQUESTED** [review]

File: example/example.cpp (hunk 2 above)
Comment (alexdevteam) [reply example/example.cpp:17]: No, it's just an example position and velocity function

File: example/example.cpp (hunk 3 above)
Comment (alexdevteam) [reply example/example.cpp:69]: Carriage return. Makes it rerender the last line basically

File: src/impl_al.hpp (hunk 4 above)
Comment (alexdevteam) [reply src/impl_al.hpp:50]: Oops, that was left from a debugging session, removing later

File: example/example.cpp (hunk 2 above)
Comment (karnkaul) [reply example/example.cpp:17]: Whoops, my bad, I didn't notice the file. 😅

**Comment by alexdevteam** [comment]
I'll need to rebase

**Review by karnkaul — APPROVED** [review]

**Pull request merged**